### PR TITLE
changes the request header for v3

### PR DESCRIPTION
**Pull request by shifali19**

### CODE REVIEW

File: keystone.go [hunk 1]
```diff
@@ -105,21 +125,34 @@ func (kClient *KeystoneClient) Authenticate() error {
 	}
 	url += "tokens"
 
+	parts := strings.Split(url, "/")
```
Comment (kaweue) [inline keystone.go]: I think we should handle situation when the input is not valid

**Review by kaweue — CHANGES_REQUESTED** [review]

File: keystone.go [hunk 2]
```diff
@@ -180,20 +301,25 @@ func (kClient *KeepaliveKeystoneClient) AddAuthentication(req *http.Request) err
...
 	return kClient.KeystoneClient.AddAuthentication(req)
 }
 
 // AddAuthentication adds the authentication data to the HTTP header.
 func (kClient *KeystoneClient) AddAuthentication(req *http.Request) error {
-	if kClient.current == nil {
-		err := kClient.Authenticate()
-		if err != nil {
-			return err
+	if len(kClient.tokenID) == 0 {
```
Comment (kaweue) [inline keystone.go]: kClient.tokenID == "" is more in golang way (In my opinion)

File: keystone.go [hunk 3]
```diff
@@ -43,6 +46,43 @@ type KeystoneToken struct {
 	Issued_At string
 }
 
+type KeystoneTokenv3 struct {
```
Comment (kaweue) [inline keystone.go:49]: You don't need a full structure. You need only fields that you would use. So I think that only IssueAt and ExpiresAT

File: keystone.go [hunk 4]
```diff
@@ -51,23 +91,101 @@ func NewKeystoneClient(auth_url, tenant_name, username, password, token string)
...
 func NewKeepaliveKeystoneClient(auth_url, tenant_name, username, password, token string) *KeepaliveKeystoneClient {
-	return &KeepaliveKeystoneClient {
+	return &KeepaliveKeystoneClient{
 		KeystoneClient{
 			auth_url,
 			tenant_name,
 			username,
 			password,
 			token,
-			nil,
```
Comment (kaweue) [inline keystone.go:66]: I suggest  using named fields in initialisation then adding empty values

**Review by kaweue — CHANGES_REQUESTED** [review]

File: keystone.go (hunk 4 above)
Comment (shifali19) [reply keystone.go:66]: I was planning on not to change the existing prototypes.That's why added empty values.

File: keystone.go (hunk 2 above)
Comment (shifali19) [reply keystone.go]: Done

File: keystone.go (hunk 3 above)
Comment (shifali19) [reply keystone.go:49]: Done

**Review by kaweue — APPROVED** [review]

**Review by mfigurski — APPROVED** [review]

**Pull request merged**